pkg/filesystem/device.go

```diff
@@ -0,0 +1,40 @@
+package filesystem
+
+import (
+	"os"
+
+	"github.com/buildbarn/bb-storage/pkg/util"
+
+	"golang.org/x/sys/unix"
+)
+
+// Device stores the name, device number and mode of a device to be used
+// when creating device nodes in the input root dev directory
+type Device struct {
+	Name string
+	Rdev deviceNumber
+	Mode os.FileMode
+}
+
+// CreateDev will take a directory and a list of devices and make nodes
+// of those devices in the /dev subdirectory of the given directory
+func CreateDev(inputRootDirectory Directory, devices []Device) error {
```

Comment:
I don’t think we should put code like this in this specific package. This is too strongly related to a certain use case. This should likely be somewhere in bb-remote-execution?